### PR TITLE
Add example for object filter on object.

### LINKED ISSUE #11762: Document Object Element Dereference Syntax
### Code of Conduct

- [X] I have read and agree to the GitHub Docs project's [Code of Conduct](https://github.com/github/docs/blob/main/CODE_OF_CONDUCT.md)

### What article on docs.github.com is affected?

https://docs.github.com/en/actions/learn-github-actions/expressions#object-filters

### What part(s) of the article would you like to see updated?

The section on object filters can be read to give the impression that the `*` syntax can only be used on arrays, but it can also be used on objects. Consider adding an example illustrating the use and behavior of `*` with objects.

### Additional information

For example, as documented by [slack-templates](https://github.com/ScribeMD/slack-templates#summarize-the-results-of-an-entire-workflow), `${{ needs.*.result }}` evaluates to an array of the results of the jobs that the current job directly depends on even though `needs` is an object, not an array. Disclosure: I am the author of slack-templates. The existing documentation resulted in a bug in Actionlint: rhysd/actionlint#66.

### PR DESCRIPTION
**Pull request by Kurt-von-Laven**
There was already an example for using an object filter on an array. Clarify that object filters can be used on objects as well and that the result is an array, not an object.

### Why:

Closes #11762.

### What's being changed:

Content only; refer to [rich diff](https://github.com/github/docs/pull/15697/files?short_path=05984cf#diff-05984cf9c2300b29e68c1644f226f65c76ea7893bfbeda57f48dda294c653ec6).

### Check off the following:

- [x] I have reviewed my changes in staging (look for "Automatically generated comment" and click **Modified** to view your latest changes).
- [x] For content changes, I have completed the [self-review checklist](https://github.com/github/docs/blob/main/contributing/self-review.md#self-review).

### Writer impact (This section is for GitHub staff members only):

- [ ] This pull request impacts the contribution experience
  - [ ] I have added the 'writer impact' label
  - [ ] I have added a description and/or a video demo of the changes below (e.g. a "before and after video")

### CODE REVIEW

**Comment by ramyaparimi** [comment]
@Kurt-von-Laven 
Thanks so much for opening a PR! I'll get this triaged for review :zap:

**Review by Abnoagstic — COMMENTED** [review]
[Aproved](https://techcrunch.com/2021/11/03/squares-cash-app-opens-up-to-teens-ages-13-to-17-with-parental-oversight/)

File: content/actions/learn-github-actions/expressions.md [hunk 1]
```diff
@@ -366,4 +366,42 @@ For example, consider an array of objects named `fruits`.
 ]
 ```
 
-The filter `fruits.*.name` returns the array `[ "apple", "orange", "pear" ]`
+The filter `fruits.*.name` returns the array `[ "apple", "orange", "pear" ]`.
+
+You may also use the `*` syntax on an object.
+
+For example, suppose you have an object named `vegetables`.
```
Comment (ethomson) [inline content/actions/learn-github-actions/expressions.md:373]:
```suggestion
You may also use the `*` syntax on an object.  For example, suppose you have an object named `vegetables`.
```

**Comment by ethomson** [comment]
Great addition @Kurt-von-Laven - thanks much.  I made one minor suggestion but this is 👍

File: content/actions/learn-github-actions/expressions.md (hunk 1 above)
Comment (Kurt-von-Laven) [reply content/actions/learn-github-actions/expressions.md]: Suggestion incorporated!

**Comment by Kurt-von-Laven** [comment]
Thank you for the review, @ethomson!

**Comment by Kurt-von-Laven** [comment]
I believe the build failure is unrelated to this pull request.

**Comment by Kurt-von-Laven** [comment]
I rebased on the latest version of main, and this time all of the checks passed. I suspect the previous failure may have been caused by a flaky test.

**Comment by ramyaparimi** [comment]
@Kurt-von-Laven We are looking into the fix. I will get this merged once the tests are passing. Thanks so much for your patience 💖

**Comment by Kurt-von-Laven** [comment]
Sounds good. The flaky test theory is supported by the fact that the tests sometimes fail and sometimes succeed when all I am doing is rebasing on the latest changes.

**Comment by Kurt-von-Laven** [comment]
@ethomson, I rebased on the latest main, and now all tests are passing.

**Review by mchammer01 — APPROVED** [review]
LGTM from Docs ✨

File: content/actions/learn-github-actions/expressions.md [hunk 2]
```diff
@@ -366,4 +366,40 @@ For example, consider an array of objects named `fruits`.
...
+  },
+  "artichokes":
+  {
+    "colors": ["green", "purple", "red", "black"],
+    "ediblePortions": ["hearts", "stems", "leaves"],
+  },
+}
+```
+
+The filter `vegetables.*.ediblePortions` could evaluate to:
```
Comment (sale6m10) [inline content/actions/learn-github-actions/expressions.md:394]: Duplicate of # #16563

**Comment by madgbec** [comment]
Please process

**Pull request merged**